sale_order_invoicing_qty_percentage/models/sale_order.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2023 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order.line"
+
+    def _prepare_invoice_line(self, **optional_values):
+        """If invoicing by quantity percentage, modify quantities."""
+        res = super()._prepare_invoice_line(**optional_values)
+        if self.env.context.get("qty_percentage"):
+            res["quantity"] *= self.env.context["qty_percentage"]
```

Comment:
Shouldn't we use float_round to avoid potential rounding issues?

Reply:
AFAIK, on create, values are rounded automatically by the precision unit, so no need to do that.